src/kirby/components/list/directives/infinite-scroll.directive.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  Directive,
+  AfterViewInit,
+  OnDestroy,
+  HostListener,
+  EventEmitter,
+  Input,
+  Output,
+  ElementRef,
+  Inject,
+} from '@angular/core';
+import { Subject } from 'rxjs';
+import { debounceTime, takeUntil, filter } from 'rxjs/operators';
+
+import { Scroll } from './scroll.model';
+import { WINDOW_PROVIDER, WINDOW_REF, WindowRef } from '../../shared/window-ref/window-ref.service';
+
+/**
+ * Specify debounce duration in ms
+ */
+export const INFINITE_SCROLL_DEBOUNCE = 100;
+
+@Directive({
+  selector: '[kirbyInfiniteScroll]',
+  providers: [WINDOW_PROVIDER],
+})
+export class InfiniteScrollDirective implements AfterViewInit, OnDestroy {
+  /**
+   * Event that will be triggered when the user has scrolled to
+   * bottom of the element
+   */
+  @Output() scrollEnd = new EventEmitter<void>();
+
+  /**
+   * If true then {@link scrollEnd} event should NOT be emitted
+   */
+  @Input() disabled = false;
+
+  /**
+   * Emits a new value on element scroll event
+   */
+  scroll$: Subject<Scroll> = new Subject<Scroll>();
+
+  /**
+   * Completes on component destroy lifecycle event
+   * used to unsubscribe from infinite observables
+   */
+  private ngUnsubscribe$ = new Subject<void>();
+
+  /**
+   * An offset from the bottom of the element to trigger {@link scrollEnd} event
+   */
+  private offset = 0.8;
+
+  constructor(private elementRef: ElementRef, @Inject(WINDOW_REF) private windowRef: WindowRef) {}
+
+  ngAfterViewInit(): void {
+    /**
+     * Subscribe to {@link scroll$} observable and emit {@link scrollEnd} event
+     * when element scroll position has surpassed the offset.
+     */
+    this.scroll$
+      .pipe(
+        takeUntil(this.ngUnsubscribe$),
+        debounceTime(INFINITE_SCROLL_DEBOUNCE),
+        filter(() => !this.disabled),
+        filter((scroll) => {
+          return (
+            scroll.elementHeight * (1 - this.offset) >=
+            scroll.distanceToViewBottom - scroll.viewHeight
+          );
+        })
+      )
+      .subscribe(() => {
+        this.scrollEnd.emit();
+      });
+  }
+
+  /**
+   * On element scroll event emit next {@link scroll$} observable value
+   */
+  @HostListener('window:scroll')
+  onScroll(): void {
+    const element = this.elementRef.nativeElement as HTMLElement;
+    const boundindClientRect = element.getBoundingClientRect();
+
+    const distanceToViewBottom = boundindClientRect.bottom;
+    const elementHeight = boundindClientRect.height;
+    const viewHeight = this.windowRef.nativeWindow.innerHeight;
```

Comment:
Why do we need this? Shouldn't it be the nearest scrollable container instead thereby removing the need for `WindowRef`?